### PR TITLE
handle the get count filters ids

### PR DESCRIPTION
**Pull request by ghazwarhili**

### CODE REVIEW

File: src/main/java/org/gridsuite/sensitivityanalysis/server/SensitivityAnalysisController.java [hunk 1]
```diff
@@ -97,6 +98,16 @@ public ResponseEntity<UUID> runAndSave(@Parameter(description = "Network UUID")
         return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(resultUuid);
     }
 
+    @PostMapping(value = "/networks/{networkUuid}/computation-count", produces = APPLICATION_JSON_VALUE)
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/sensitivityanalysis/server/SensitivityAnalysisController.java:101]: /networks/{networkUuid}/parameters/complexity ?

File: src/main/java/org/gridsuite/sensitivityanalysis/server/SensitivityAnalysisController.java [hunk 2]
```diff
@@ -97,6 +98,16 @@ public ResponseEntity<UUID> runAndSave(@Parameter(description = "Network UUID")
...
+    @Operation(summary = "Get computation count of the total of identifiables for a list of filters")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The sensitivity analysis identifiables count"),
+        @ApiResponse(responseCode = "404", description = "Sensitivity analysis identifiables count result has not been found")})
+    public ResponseEntity<Integer> getComputationCount(@Parameter(description = "Result UUID") @PathVariable("networkUuid") UUID networkUuid,
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/sensitivityanalysis/server/SensitivityAnalysisController.java:105]: description = "Network UUID"

File: src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisInputBuilderService.java [hunk 3]
```diff
@@ -113,6 +114,18 @@ private List<IdentifiableAttributes> goGetIdentifiables(EquipmentsContainer filt
         }
     }
 
+    private Integer getComputationCount(Map<String, List<UUID>> ids, UUID networkUuid, Integer containersAttributesCount) {
+        Map<String, List<Integer>> map = filterService.getIdentifiablesCount(ids, networkUuid, null);
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisInputBuilderService.java:118]: for contingency list ids, you should request the actions-server and not the filter-server, to get all the contingencies ...

File: src/test/java/org/gridsuite/sensitivityanalysis/server/SensitivityAnalysisControllerTest.java [hunk 4]
```diff
@@ -465,7 +469,7 @@ public void setUp() {
         given(filterService.getIdentifiablesFromFilter(EQUIPMENTS_IN_VOLTAGE_REGULATION_FILTERS_UUID, NETWORK_UUID, VARIANT_2_ID)).willReturn(EQUIPMENTS_IN_VOLTAGE_REGULATION);
         given(filterService.getIdentifiablesFromFilter(EQUIPMENTS_IN_VOLTAGE_REGULATION_FILTERS_UUID, NETWORK_UUID, null)).willReturn(EQUIPMENTS_IN_VOLTAGE_REGULATION);
         given(filterService.getIdentifiablesFromFilter(EQUIPMENTS_IN_VOLTAGE_REGULATION_FILTERS_UUID, NETWORK_STOP_UUID, VARIANT_2_ID)).willReturn(EQUIPMENTS_IN_VOLTAGE_REGULATION);
-
+        given(filterService.getIdentifiablesCount(IDS, NETWORK_UUID, null)).willReturn(Map.of("0", List.of(6), "1", List.of(6), "2", List.of(6)));
```
Comment (FranckLecuyer) [inline src/test/java/org/gridsuite/sensitivityanalysis/server/SensitivityAnalysisControllerTest.java:472]: See remark above for contingencies ...

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/sensitivityanalysis/server/SensitivityAnalysisController.java (hunk 1 above)
Comment (ghazwarhili) [reply src/main/java/org/gridsuite/sensitivityanalysis/server/SensitivityAnalysisController.java:101]: It's a proposal from Slimane

File: src/main/java/org/gridsuite/sensitivityanalysis/server/SensitivityAnalysisController.java (hunk 2 above)
Comment (ghazwarhili) [reply src/main/java/org/gridsuite/sensitivityanalysis/server/SensitivityAnalysisController.java]: done

File: src/test/java/org/gridsuite/sensitivityanalysis/server/SensitivityAnalysisControllerTest.java (hunk 4 above)
Comment (ghazwarhili) [reply src/test/java/org/gridsuite/sensitivityanalysis/server/SensitivityAnalysisControllerTest.java]: done

File: src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisInputBuilderService.java (hunk 3 above)
Comment (ghazwarhili) [reply src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisInputBuilderService.java:131]: done

File: src/main/java/org/gridsuite/sensitivityanalysis/server/service/FilterService.java [hunk 5]
```diff
@@ -53,4 +58,25 @@ public List<IdentifiableAttributes> getIdentifiablesFromFilter(UUID uuid, UUID n
...
+        var path = uriComponentsBuilder
+                .build()
+                .toUriString();
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+
+        HttpEntity<Map<String, List<UUID>>> httpEntity = new HttpEntity<>(ids, headers);
+
+        return restTemplate.exchange(filterServerBaseUri + path, HttpMethod.POST, httpEntity,
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/sensitivityanalysis/server/service/FilterService.java:78]: HttpMethod.GET

File: src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisInputBuilderService.java [hunk 6]
```diff
@@ -430,6 +456,16 @@ private void buildSensitivityNodes(SensitivityAnalysisRunContext context, Networ
...
     }
 
+    public Integer getComputationCount(Map<String, List<UUID>> ids, UUID networkUuid, String variantId, Boolean isInjectionsSet) {
+        Integer containersAttributesCount = 1;
+        if (Boolean.TRUE.equals(isInjectionsSet)) {
+            containersAttributesCount *= ids.get(INJECTIONS).size();
+            ids.remove(INJECTIONS);
+        }
+        containersAttributesCount = getComputationCount(ids, networkUuid, variantId, containersAttributesCount);
+        return containersAttributesCount;
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisInputBuilderService.java:466]: `return getComputationCount(ids, networkUuid, variantId, containersAttributesCount) `?

File: src/test/java/org/gridsuite/sensitivityanalysis/server/SensitivityAnalysisControllerTest.java [hunk 7]
```diff
@@ -133,6 +133,14 @@ public class SensitivityAnalysisControllerTest {
     private static final UUID MONITORED_VOLTAGE_LEVELS_FILTERS_NODES_UUID = UUID.randomUUID();
     private static final UUID EQUIPMENTS_IN_VOLTAGE_REGULATION_FILTERS_UUID = UUID.randomUUID();
     private static final UUID CONTINGENCIES_NODES_UUID = UUID.randomUUID();
+    private static final List<UUID> MONITORED_BRANCHES_FILTERS_UUID = List.of(UUID.randomUUID());
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/sensitivityanalysis/server/SensitivityAnalysisControllerTest.java:136]: Really need these constants ?

**Review by SlimaneAmar — APPROVED** [review]

**Review by FranckLecuyer — APPROVED** [review]

**Review by SlimaneAmar — APPROVED** [review]

**Comment by ghazwarhili** [comment]
add and handle the sensitivity analysis factors count
Signed-off-by: Ghazwa REHILI <ghazwa.rehili@rte-france.com>

**Pull request merged**